pkg/proto/nbcontract/v1/customlinuxosconfig.proto

```diff
@@ -0,0 +1,55 @@
+syntax = "proto3";
+package nbcontract.v1;
+
+message CustomLinuxOSConfig {
+    // The size in MB of a swap file that will be created on each node
+    optional int32 swap_file_size = 1;
+  
+    // Valid values are "always", "defer", "defer+madvise", "madvise" and "never"
+    optional string transparent_hugepage_support = 2;
+  
+    // Valid values are "always", "madvise" and "never"
+    optional string transparent_defrag = 3;
+  
+    // Sysctl settings for Linux agent nodes
+    optional SysctlConfig sysctl_config = 4;
+  
+    // Ulimit settings for Linux agent nodes
+    optional UlimitConfig ulimit_config = 5;
+  }
+
+message SysctlConfig {
+    optional int32 NetCoreSomaxconn = 1;
+    optional int32 NetCoreNetdevMaxBacklog = 2;
+    optional int32 NetCoreRmemDefault = 3;
+    optional int32 NetCoreRmemMax = 4;
+    optional int32 NetCoreWmemDefault = 5;
+    optional int32 NetCoreWmemMax = 6;
+    optional int32 NetCoreOptmemMax = 7;
+    optional int32 NetIpv4TcpMaxSynBacklog = 8;
+    optional int32 NetIpv4TcpMaxTwBuckets = 9;
+    optional int32 NetIpv4TcpFinTimeout = 10;
+    optional int32 NetIpv4TcpKeepaliveTime = 11;
+    optional int32 NetIpv4TcpKeepaliveProbes = 12;
+    optional int32 NetIpv4TcpkeepaliveIntvl = 13;
+    optional bool NetIpv4TcpTwReuse = 14;
+    optional string NetIpv4IpLocalPortRange = 15;
+    optional int32 NetIpv4NeighDefaultGcThresh1 = 16;
+    optional int32 NetIpv4NeighDefaultGcThresh2 = 17;
+    optional int32 NetIpv4NeighDefaultGcThresh3 = 18;
+    optional int32 NetNetfilterNfConntrackMax = 19;
+    optional int32 NetNetfilterNfConntrackBuckets = 20;
+    optional int32 FsInotifyMaxUserWatches = 21;
+    optional int32 FsFileMax = 22;
+    optional int32 FsAioMaxNr = 23;
+    optional int32 FsNrOpen = 24;
+    optional int32 KernelThreadsMax = 25;
+    optional int32 VMMaxMapCount = 26;
+    optional int32 VMSwappiness = 27;
+    optional int32 VMVfsCachePressure = 28;
+  }
+  
+  message UlimitConfig {
```

Comment:
This is currently not used in aks-rp, however long term plan is to eventually use it and expand it with a lot more params (possibly allow users to specify arbitrary ulimits). Not sure if its worth it to include it at this point?